### ISSUE TITLE
Placeholder should have a title for WCAG AA compliance

### ISSUE THREAD

**Issue by onesneakymofo** [select2/select2#5963]
Edit:
The information below is incorrect. @kevin-brown provided the correct description below.

Original:
Hello,

I upgraded from 3.x to 4.0.13 recently, and it caused my test suite to fail as far as WCAG compliance goes.

Here's a bin:

https://jsbin.com/sakexad/edit?html,js,output

Here's the failure I'm getting when I run my test suite using Deque's WCAG tool:

 ``` 
     1) aria-input-field-name: ARIA input fields have an accessible name (serious)
           https://dequeuniversity.com/rules/axe/3.3/aria-input-field-name?application=axeAPI
           The following 1 node violates this rule:

               Selector: 
               #select2-example1-container
               
               HTML: 
               <span class="select2-selection__rendered" id="select2-example1-container" role="textbox" aria-readonly="true">
                 <span class="select2-selection__placeholder">Hello</span>
               </span>
               
               Fix any of the following:
               - aria-label attribute does not exist or is empty
               - aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
               - Element has no title attribute or the title attribute is empty
```

With the placeholder, you get the following:
```
<span class="select2-selection__rendered" id="select2-example1-container" role="textbox" aria-readonly="true">
  <span class="select2-selection__placeholder">
    Hello
  </span>
</span>
```

If you remove the placeholder, you get the following (https://jsbin.com/quruziz/edit?html,js,output):

```
<span class="select2-selection__rendered" id="select2-example1-container" role="textbox" aria-readonly="true" title="Hello">Hello</span>
```

According to the WCAG guidelines, a referenced aria-labelledby should not have child elements.  Removing the additional span should solve this.

Also, I tried the beta releases, and they did not fix it unfortunately.

Thanks.

**Comment by kevin-brown** [comment]
As far as I can tell, the issue is not the added down. Since we don't appear to be setting `aria-labelledby` anywhere, it's not interfering with any label calculations. The issue is that placeholders do not set the `title` attribute.

When you are not using a placeholder, the first item I'd automatically selected by the browser and Select2 displays the text of the selected item in the `title` attribute. So you're not getting the WCAG error because the `title` is being set, not because it's missing spans.

I can look into setting the title on the container to the placeholder value.

**Comment by onesneakymofo** [comment]
@kevin-brown `aria-labelledby` is utilized:

```
<span class="select2-selection select2-selection--single" role="combobox" aria-haspopup="true" aria-expanded="false" tabindex="0" aria-disabled="false" aria-labelledby="select2-example1-container">
  <span class="select2-selection__rendered" id="select2-example1-container" role="textbox" aria-readonly="true">
    <span class="select2-selection__placeholder">Hello</span>
    </span>
    <span class="select2-selection__arrow" role="presentation">
      <b role="presentation"></b>
    </span>
</span>
```

It points to the `#select2-example1-container` element I posted above.  The issue is that the first child in `#select2-example1-container` is a span and not a text value. This is what breaks compliance.

With that said, a title should be sufficient.  Let me know once it's ready for testing, and I'll QA.

Thanks again.

**Comment by kevin-brown** [comment]
Are you able to point me to where it says the label can't have nested elements?

**Comment by onesneakymofo** [comment]
@kevin-brown I'm sorry.  It can have nested elements according to W3C. I'm looking at it from the wrong point of view - I thought the `aria-labelledby` was referencing the placeholder itself, but it's referencing all of the options instead.
